src/rust/src/x509/certificate.rs

```diff
@@ -731,6 +731,103 @@ pub(crate) fn parse_access_descriptions(
     Ok(ads.to_object(py))
 }
 
+fn parse_naming_authority<'p>(
+    py: pyo3::Python<'p>,
+    authority: NamingAuthority<'p>,
+) -> CryptographyResult<pyo3::PyObject> {
+    let py_id = match &authority.id {
+        Some(data) => oid_to_py_oid(py, data)?.to_object(py),
+        None => py.None(),
```

Comment:
Instead of doing `to_object()` on the some path, I'd recommend doing a `bind(py)` on the `None` path. It's a moderately more flexible pattern.

Same for the `py_url` path.